conans/test/integration/toolchains/cmake/cmakedeps/test_cmakedeps.py

```diff
@@ -707,3 +707,35 @@ def generate(self):
     assert 'set(dep_NO_SONAME_MODE_RELEASE TRUE)' in dep
     other = c.load("app/other-release-data.cmake")
     assert 'set(other_other_mycomp1_NO_SONAME_MODE_RELEASE TRUE)' in other
+
+def test_cmakedeps_set_legacy_variable_name():
+    client = TestClient()
+    conanfile = str(GenConanfile("dep", "1.0"))
+    conanfile += """
+    def package_info(self):
+        self.cpp_info.set_property("cmake_file_name", "CMakeFileName")
+    """
+    client.save({"dep/conanfile.py": conanfile})
+    client.run("create dep")
+    client.save({"conanfile.py": GenConanfile("lib", "1.0")
+                .with_requires("dep/1.0").with_generator("CMakeDeps").with_setting("build_type")})
+    client.run("install .")
```

Comment:
you could simplify this a bit too:

```suggestion
    client.run("install --requires=dep/1.0 -g CMakeDeps")
```